freqtrade/optimize/edge.py

```diff
@@ -0,0 +1,1037 @@
+
+                # create a new dict
+                close_index: int = t_exit_ind
+                bslap_result = {}  # Must have at start or we end up with a list of multiple same last result
+                bslap_result["pair"] = pair
+                bslap_result["stoploss"] = stop
+                bslap_result["profit_percent"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["profit_abs"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["open_time"] = np_bslap_dates[t_open_ind + 1]  # use numpy array, pandas 20x slower
+                bslap_result["close_time"] = np_bslap_dates[close_index]  # use numpy array, pandas 20x slower
+                bslap_result["open_index"] = t_open_ind + 1  # +1 as we buy on next.
+                bslap_result["close_index"] = close_index
+                bslap_result["trade_duration"] = ""  # To be 1 vector calc across trades when loop complete
+                bslap_result["open_at_end"] = False
+                bslap_result["open_rate"] = round(np_trade_enter_price, 15)
+                bslap_result["close_rate"] = round(np_trade_exit_price, 15)
+                bslap_result["exit_type"] = t_exit_type
+                bslap_result["sell_reason"] = t_exit_type #duplicated, but I don't care
+                # append the dict to the list and print list
+                bslap_pair_results.append(bslap_result)
+
+                if t_exit_type is SellType.STOP_LOSS:
+                    stop_stops_count = stop_stops_count + 1
+
+                if debug:
+                    print("The trade dict is: \n", bslap_result)
+                    print("Trades dicts in list after append are: \n ", bslap_pair_results)
+
+                """
+                Loop back to start. t_exit_last becomes where loop
+                will seek to open new trades from.
+                Push index on 1 to not open on close
+                """
+                t_exit_last = t_exit_ind + 1
+
+            if debug_timing:
+                t_t = f(st)
+                print("8+trade", str.format('{0:.17f}', t_t))
+
+        # Send back List of trade dicts
+        return bslap_pair_results
+    
+    def _process_result(self, data: Dict[str, Dict], results: DataFrame, stoploss_range) -> str:
+        """
+        This is a temporary version of edge positioning calculation. 
+        The function will be eventually moved to a plugin called Edge in order to calculate necessary WR, RRR and 
+        other indictaors related to money management periodically (each X minutes) and keep it in a storage. 
+        The calulation will be done per pair and per strategy. 
+        """
+
+        # Removing open trades from dataset
+        results = results[results.open_at_end == False]
+        ###################################
+
+        # Removing pairs having less than min_trades_number
+        min_trades_number = 50
```

Comment:
This is an arbitrarily high number of backtest results to find for the stats to be valid.
On a 4 hour frame this could take the user back to a very different trend-time whih many provide counter results. 
Perhaps this should be driven from config file at users discretion